core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -1368,6 +1369,14 @@ aot_compile_op_atomic_wait(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
 
     PUSH_I32(ret_value);
 
+#if WASM_ENABLE_THREAD_MGR != 0
```

Comment:
We also need this change in `aot_emit_function.c`, as in #1975.

Reply:
As the main thread's exception won't be cleared by sub thread, main thread will exit after returning the function to the caller, and won't fetch the `unreachable` opcode and throw "unreachable" exception again. It works for the test cases, for the performance reason, I didn't add the check for AOT.

Reply:
The change in `aot_emit_function.c` is not needed for the main thread, but for the case in which a spawned thread is waiting on a `poll_oneoff`. You can reproduce it with this example https://github.com/WebAssembly/wasi-threads/blob/main/test/testsuite/wasi_threads_exit_main_wasi.wat. We want to exit after the `poll_oneoff` gets interrupted without fetching the next instruction (that is `unreachable`) https://github.com/WebAssembly/wasi-threads/blob/c1b421c2158b745c840b5bdca98ed870c2a6d11a/test/testsuite/wasi_threads_exit_main_wasi.wat#L26

Reply:
OK, I added the checks in aot_emit_function.c.